### PR TITLE
Add Unit tests for Utils module

### PR DESCRIPTION
**Pull request by wshao08**

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/utils/MemUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    sizeSum += 8 + TSDataType.INT64.getDataTypeSize();
+    dataTypes.add(TSDataType.FLOAT.ordinal());
+    sizeSum += 8 + TSDataType.FLOAT.getDataTypeSize();
+    dataTypes.add(TSDataType.DOUBLE.ordinal());
+    sizeSum += 8 + TSDataType.DOUBLE.getDataTypeSize();
+    dataTypes.add(TSDataType.TEXT.ordinal());
+    sizeSum += TSDataType.TEXT.getDataTypeSize();
+    InsertTabletPlan insertPlan = new InsertTabletPlan(device, measurements, dataTypes);
+    Assert.assertEquals(sizeSum, MemUtils.getRecordSize(insertPlan, 0, 1, false));
+  }
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/MemUtilsTest.java:58]: What about seperating this single test to two? I think it's better to keep the test atomic.

File: server/src/test/java/org/apache/iotdb/db/utils/MemUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  @Test
+  public void TsRecordSizeTest() {
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/MemUtilsTest.java:61]: `TsRecordSizeTest` -> `TsRecordMemTest`. And add javadoc for this test, it tests getDataPointMem() and getTsRecordMem().

File: server/src/test/java/org/apache/iotdb/db/utils/MemUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    Assert.assertEquals(MemUtils.getStringMem("s4") + 24, MemUtils.getDataPointMem(point4));
+    totalSize += MemUtils.getDataPointMem(point4);
+    record.addTuple(point4);
+
+    DataPoint point5 = new BooleanDataPoint("s5", true);
+    Assert.assertEquals(MemUtils.getStringMem("s5") + 17, MemUtils.getDataPointMem(point5));
+    totalSize += MemUtils.getDataPointMem(point5);
+    record.addTuple(point5);
+
+    totalSize += 8 * record.dataPointList.size() + MemUtils.getStringMem(device)+ 16;
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/MemUtilsTest.java:91]:
```suggestion
    totalSize += 8 * record.dataPointList.size() + MemUtils.getStringMem(device) + 16;
```

File: server/src/test/java/org/apache/iotdb/db/utils/MemUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    DataPoint point4 = new DoubleDataPoint("s4", 1.0d);
+    Assert.assertEquals(MemUtils.getStringMem("s4") + 24, MemUtils.getDataPointMem(point4));
+    totalSize += MemUtils.getDataPointMem(point4);
+    record.addTuple(point4);
+
+    DataPoint point5 = new BooleanDataPoint("s5", true);
+    Assert.assertEquals(MemUtils.getStringMem("s5") + 17, MemUtils.getDataPointMem(point5));
+    totalSize += MemUtils.getDataPointMem(point5);
+    record.addTuple(point5);
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/MemUtilsTest.java:89]: Why not check StringDataPoint?

File: server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+import org.apache.iotdb.tsfile.read.filter.TimeFilter;
+import org.apache.iotdb.tsfile.read.filter.basic.Filter;
+import org.apache.iotdb.tsfile.utils.Binary;
+import org.apache.iotdb.tsfile.utils.TsPrimitiveType;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SerializeUtilsTest {
+  @Test
+  public void serdesStringTest() {
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java:45]: I don't think `serdes` is a good abbreviation...

File: server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+    List<String> slist = Arrays.asList("abc", "123", "y87@", "9+&d\n");
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream outputStream = new DataOutputStream(baos);
+    SerializeUtils.serializeStringList(slist, outputStream);
+    ByteBuffer buffer = ByteBuffer.wrap(baos.toByteArray());
+    Assert.assertEquals(slist, SerializeUtils.deserializeStringList(buffer));
+  }
+
+  @Test
+  public void serdesIntListTest() {
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java:65]: Hi, I checked the name of this method in SerializeUtils, which called just serialize. FMP, it's better to rename it.
`serialize` -> `serializeIntList`,  `serialize` -> `serializeIntSet`...

File: server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+    ByteBuffer buffer = ByteBuffer.wrap(baos.toByteArray());
+    Node anotherNode = new Node("127.0.0.1", 6667, 1, 6535, 4678);
+    SerializeUtils.deserialize(anotherNode, buffer);
+    Assert.assertEquals(node, anotherNode);
+  }
+
+  @Test
+  public void serdesINT32BatchDataTest() {
+    BatchData batchData = new BatchData(TSDataType.INT32);
+    int ivalue = 0;
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java:104]: Actually, there are three types of BatchData, and two of them are subClass of BatchData. You should test them all.

File: server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+      batchData.next();
+      anotherBatch.next();
+    }
+  }
+
+  @Test
+  public void serdesTVPairTest() {
+    List<TimeValuePair> TVPairs = new ArrayList<>();
+    TimeValuePair p1 = new TimeValuePair(0, TsPrimitiveType.getByType(TSDataType.BOOLEAN, true));
+    TVPairs.add(p1);
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java:222]: Add javadoc for this and the next test.

File: server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    DataOutputStream outputStream = new DataOutputStream(baos);
+    SerializeUtils.serializeObjects(objects, outputStream);
+    ByteBuffer buffer = ByteBuffer.wrap(baos.toByteArray());
+    Assert.assertArrayEquals(objects, SerializeUtils.deserializeObjects(buffer));
+  }
+
+  @Test
+  public void serdesLongTest() {
+    long[] array = {1, 10, 100, 1000, 10000};
```
Comment (Alima777) [inline server/src/test/java/org/apache/iotdb/db/utils/SerializeUtilsTest.java:292]: It's Longs here. And add javadoc.

**Review by Alima777 — CHANGES_REQUESTED** [review]

**Comment by Alima777** [comment]
Hi, check the CI please.

**Review by Alima777 — APPROVED** [review]
LGTM expect the test named `serdes...Test`... What do you think? @qiaojialin

**Comment by qiaojialin** [comment]
Hi, serDe is ok~

**Pull request merged**